twos-complement/level-protocol-website
# Adding members and communities counters + integration + backfilling

## Purpose

Adding new queries to keep track of communities and members' counts. See: https://app.dework.xyz/indiedao/lvl-protocol?taskId=07a1f130-d5a8-4f99-9eaf-309af54eb5a4

## How to setup

- Go to Fauna Dashboard and replace the schema
- Run the newly added shell instructions, see README for more details on how to do this.

## How to test

- You can go into Fauna Dashboard and run the queries yourself from `util/queries.js` 
- Additionally, you may want to actually create members from the app to make sure the counter is updated.

## How to backfill counters in development and production?

- Enable `NEXT_PUBLIC_ENABLE_DATA_MIGRATIONS` in the corresponding environment you want to fix things.
- Call `GET api/backfill-data` endpoint to update counters, you should get a response like this:

```json
{
   "success":true,
   "data":{
      "previousMembersCount":0,
      "previousCommunitiesCount":0,
      "membersCount":0,
      "communitiesCount":2
   }
}
```

- You could also do this locally by setting up the FAUNA_SECRET value to the one from dev or prod.
